src/context.rs

```diff
@@ -268,14 +321,28 @@ unsafe fn cb_get_config(
     key: *const libc::c_char,
     def: *const libc::c_char,
 ) -> *mut libc::c_char {
-    dc_sqlite3_get_config(context, &context.sql.clone().read().unwrap(), key, def)
+    let d = if def.is_null() {
+        None
+    } else {
+        Some(as_str(def))
+    };
+    let res = dc_sqlite3_get_config(
+        context,
+        &context.sql.clone().read().unwrap(),
+        as_str(key),
+        d,
+    );
+    if let Some(res) = res {
```

Comment:
Isn't match more idiomatic than an if let with and else clause?

Reply:
I don't think there is a strong agreement.